### ISSUE TITLE
Dismissing file dialog in Android 2.3.x without selecting a file result in an empty file submission

### ISSUE THREAD

**Issue by rnicholus** [FineUploader/fine-uploader#998]
Reported via Stack Overflow at http://stackoverflow.com/questions/18854655/file-with-no-name-or-size-in-fineuploader.

Android 2.3.x fires the change event on a file input element even if the file dialog is dismissed without selecting a file.  In this case, the value of the input element is an empty string.  We should probably work around this by ignoring all change events associated with non File API browsers that contains an empty string for the file input value.
### Workaround

The workaround is simple but a bit kludgey.  This will no longer be necessary once Fine Uploader has been patched to work around this issue internally.  Until then...

TL;DR
Contribute a [validate event handler](http://docs.fineuploader.com/api/events.html#validate):

``` javascript
$("#uploader").fineUploader().on("validate", function(event, fileOrBlobData) {
    return fileOrBlobData.name.length > 0 && fileOrBlobData.name !== "qqfile";
});
```

The above code will reject any file with an empty name.
#### The long explanation for this workaround

The handler should return false for all files with a name of "qqfile".  One would think that your handler should only return false (cancelling the upload) for all files with an empty name, but there is some logic inside Fine Uploader that, unintentionally, sets the name property of the object passed to the validate handlers to the `name` attribute of the file input element if the `value` of this element is "falsy".  There's a bit more to it than that, but see the code if you are interested.
